### PR TITLE
[RLlib] SAC algo cleanup.

### PR DESCRIPTION
**Pull request by sven1977**
SAC algo cleanup/comments/docstrings/type annotations.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ericl** [comment]
@michaelzhiluo can you review this?

File: doc/source/rllib-algorithms.rst [hunk 1]
```diff
@@ -399,15 +402,23 @@ HalfCheetah    9664                       ~7700
 Soft Actor Critic (SAC)
 ------------------------
 |pytorch| |tensorflow|
-`[paper] <https://arxiv.org/pdf/1801.01290>`__ `[implementation] <https://github.com/ray-project/ray/blob/master/rllib/agents/sac/sac.py>`__
+`[original paper] <https://arxiv.org/pdf/1801.01290>`__,
+`[follow up paper] <https://arxiv.org/pdf/1812.05905.pdf>`__,
```
Comment (michaelzhiluo) [inline doc/source/rllib-algorithms.rst:406]: Perhaps put all papers and follow ups on one line

File: doc/source/rllib-algorithms.rst [hunk 2]
```diff
@@ -399,15 +402,23 @@ HalfCheetah    9664                       ~7700
...
 
-RLlib's soft-actor critic implementation is ported from the `official SAC repo <https://github.com/rail-berkeley/softlearning>`__ to better integrate with RLlib APIs. Note that SAC has two fields to configure for custom models: ``policy_model`` and ``Q_model``.
+RLlib's soft-actor critic implementation is ported from the `official SAC repo <https://github.com/rail-berkeley/softlearning>`__ to better integrate with RLlib APIs.
+Note that SAC has two fields to configure for custom models: ``policy_model`` and ``Q_model``, the ``model`` field of the config will be ignored.
 
-Tuned examples: `Pendulum-v0 <https://github.com/ray-project/ray/blob/master/rllib/tuned_examples/sac/pendulum-sac.yaml>`__, `HalfCheetah-v3 <https://github.com/ray-project/ray/blob/master/rllib/tuned_examples/sac/halfcheetah-sac.yaml>`__
+Tuned examples:
+`Pendulum-v0 <https://github.com/ray-project/ray/blob/master/rllib/tuned_examples/sac/pendulum-sac.yaml>`__,
+`HalfCheetah-v3 <https://github.com/ray-project/ray/blob/master/rllib/tuned_examples/sac/halfcheetah-sac.yaml>`__,
+`CartPole-v0 <https://github.com/ray-project/ray/blob/master/rllib/tuned_examples/sac/cartpole-sac.yaml>`__
```
Comment (michaelzhiluo) [inline doc/source/rllib-algorithms.rst:421]: Might want to mention (Discrete)

File: doc/source/rllib-algorithms.rst (hunk 1 above)
Comment (sven1977) [reply doc/source/rllib-algorithms.rst:406]: done

File: doc/source/rllib-algorithms.rst (hunk 2 above)
Comment (sven1977) [reply doc/source/rllib-algorithms.rst:421]: done

**Comment by sven1977** [comment]
@michaelzhiluo

**Review by michaelzhiluo — APPROVED** [review]

**Pull request merged**